api/nakadi-event-bus-api.yaml

```diff
@@ -1567,13 +1651,14 @@ definitions:
           future there could be others.
       schema:
         type: string
-        $ref: '#/definitions/EventTypeSchema'
         description: |
           The schema as string in the syntax defined in the field type. Failure to respect the
           syntax will fail any operation on an EventType.
 
-          To have a generic, undefined schema it is possible to define the schema as `"schema":
-          "{\"additionalProperties\": true}"`.
+          When defining a schema, every property should be explicitly defined with its type. It means that,
+          additionalProperties true is not accepted.
```

Comment:
@conorclifford I think your example will help us better define our goal with this feature. I think the principle of open for extension, closed for change would fit nicely here.

I see no problems with simple `object` properties. Yes, it's extensible, since you can always add something more inside it. But if we allow further changes to this `object`, by defining some property, it changes the previously defined "contract". Let me give you an example:

```javascript
{
  "properties": {
    "extensible_object": {
      "type": "object";
     }
  }
}
```

And later, it changes to 

```javascript
{
  "properties": {
    "extensible_object": {
      "type": "object";
      "properties": {
        "some_very_specific_constraint": {
          "type": "string";
        }
     }
  }
}
```

What it means is that, for some time, this event type might have accepted `some_very_specific_constraint` property with any `type`, but after some point, it defined that the `type` is a string.

That would not fulfil one of the goals - [backward compatibility](https://docs.google.com/document/d/1FmGHxI9oklxN16LKZe1UG_U21jtQFVCtLtRiC2W1pAk/edit#heading=h.hov7tfyvg4a9).

That all said, I think things are clearer now. What we need to guarantee here is there is an "open definition" it should never change.

Just to make it very clear: if "additionalProperties" is allowed, it might not be able to change this part of the schema in the future.

Would it work for you?